crates/pico/tests/arg_reference.rs

```diff
@@ -0,0 +1,79 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+use calc::{ast::Program, error::Result, eval::eval, lexer::Lexer, parser::Parser};
+use pico_core::{database::Database, source::SourceId};
+use pico_macros::{memo, Source};
+
+mod calc;
+
+static VALUE: AtomicUsize = AtomicUsize::new(0);
+static EXPECTED: AtomicUsize = AtomicUsize::new(0);
+
+#[test]
+fn arg_reference() {
+    let mut db = Database::default();
+
+    let input = db.set(Input {
+        key: "input",
+        value: "2 + 2 * 2".to_string(),
+    });
+
+    let value = evaluate_input(&db, input);
+    // assert that the value was not cloned
+    assert_eq!(VALUE.load(Ordering::SeqCst), 0);
+
+    let expected = Expected(6);
+    assert_result(&db, value, expected);
+    // assert that the argument of type `&Value` was cloned only once to params store
```

Comment:
Nit: `&Value` was cloned once, when it was added to the params store.

----

Does this sort of negate the utility of using references?